File: API/src/main/java/org/sheeper/blogify/model/BlogPost.java
package org.sheeper.blogify.model;

import java.util.Date;

import com.fasterxml.jackson.annotation.JsonFormat;

public class BlogPost {

    private String title;
    private Date date;
    private String author;
    private String introduction;
    private String content;
    private String postUrl;
    private String blogUrl;
    private int page;

    public BlogPost() {

    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    @JsonFormat(pattern = "dd.MM.yyyy")
    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public String getIntroduction() {
        return introduction;
    }

    public void setIntroduction(String introduction) {
        this.introduction = introduction;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public String getBlogUrl() {
        return blogUrl;
    }

    public void setBlogUrl(String blogUrl) {
        this.blogUrl = blogUrl;
    }

    public String getPostUrl() {
        return postUrl;
    }

    public void setPostUrl(String postUrl) {
        this.postUrl = postUrl;
    }

    public int getPage() {
        return page;
    }

    public void setPage(int page) {
        this.page = page;
    }
}

File: Mobile/lib/BlogPost.dart
class BlogPost {
  final String title;
  final String date;
  final List<String> authors;
  final String summary;
  final String content;
  final String blogUrl;
  final String postUrl;

  BlogPost(
      {required this.title,
      required this.date,
      required this.authors,
      required this.summary,
      required this.content,
      required this.blogUrl,
      required this.postUrl});

  factory BlogPost.fromJson(Map<String, dynamic> json) {
    return BlogPost(
      title: json['title'],
      date: json['date'] != null ? json['date'] : 'N/A',
      authors: List<String>.from(json['authors']),
      summary: json['summary'],
      content: json['content'],
      blogUrl: json['blogUrl'],
      postUrl: json['postUrl'],
    );
  }
}
